### ISSUE TITLE
Correction: Sponge does not use Minecraft decompiled code

### ISSUE THREAD

**Issue by mastercoms** [IngotPowered/IngotEngine#12]
Sponge is a complete rewrite of the minecraft server.

**Comment by dreadiscool** [comment]
Hello,

Sponge uses Forge, which is based on MCP, which uses decompiled code

Thanks

**Comment by mastercoms** [comment]
That is one implementation of Sponge, but I guess it doesn't really matter. Good luck with this.

**Comment by alexrallen** [comment]
Correction, sponge does not rely on minecraft decompiled code. Sponge is an API, it is platform independent and does not rely on decompiled code itself. The forge coremod implementation may rely on decompiled code but it is not the only implementation and is not sponge itself.

**Comment by simon816** [comment]
Also, there's nothing wrong with Glowstone either. It is clean-room (I think) and implements the Bukkit API (no legal issues) and the Minecraft Server protocol

**Comment by mastercoms** [comment]
Yes, Glowstone is a clean room rewrite, but not a clean room rewrite in the strictest sense.
There is no legal shakiness with Glowstone or Sponge. They were made to avoid the legal shakiness of Bukkit, Sponge directly in response to a disaster relating to a flaw in Bukkit's legal structure, and Glowstone passively in response to Bukkit's reliance on decompiled code.
